app/vmui/packages/vmui/src/components/TopQueries/TopQueries.tsx

```diff
@@ -0,0 +1,139 @@
+    const durObject = durItems.reduce((prev, curr) => {
+      const dur = isSupportedDuration(curr);
+      return dur ? {...prev, ...dur} : {...prev};
+    }, {});
+    const delta = dayjs.duration(durObject).asMilliseconds();
+    return !!delta;
+  }, [maxLifetime]);
+
+  const onTopNChange = (e: ChangeEvent<HTMLTextAreaElement|HTMLInputElement>) => {
+    topQueriesDispatch({type: "SET_TOP_N", payload: +e.target.value});
+  };
+
+  const onMaxLifetimeChange = (e: ChangeEvent<HTMLTextAreaElement|HTMLInputElement>) => {
+    topQueriesDispatch({type: "SET_MAX_LIFE_TIME", payload: e.target.value});
+  };
+
+  const onApplyQuery = () => {
+    topQueriesDispatch({type: "SET_RUN_QUERY"});
+  };
+
+  const onKeyDown = (e: KeyboardEvent) => {
+    if (e.key === "Enter") onApplyQuery();
+  };
+
+  useEffect(() => {
+    if (!data) return;
+    if (!topN) topQueriesDispatch({type: "SET_TOP_N", payload: +data.topN});
+    if (!maxLifetime) topQueriesDispatch({type: "SET_MAX_LIFE_TIME", payload: data.maxLifetime});
+  }, [data]);
+
+  return (
+    <Box p={4} style={{minHeight: "calc(100vh - 64px)"}}>
+      {loading && <Spinner isLoading={true} height={"100%"}/>}
+
+      <Box boxShadow="rgba(99, 99, 99, 0.2) 0px 2px 8px 0px;" p={4} pb={2} m={-4} mb={4}>
+        <Box display={"flex"} alignItems={"flex"} mb={2}>
+          <Box mr={2} flexGrow={1}>
+            <TextField
+              fullWidth
+              label="Max lifetime"
+              size="medium"
+              variant="outlined"
+              value={maxLifetime}
+              error={!maxLifetimeValid}
+              helperText={!maxLifetimeValid ? "Invalid duration value" : `For example ${exampleDuration}`}
+              onChange={onMaxLifetimeChange}
+              onKeyDown={onKeyDown}
+            />
+          </Box>
+          <Box mr={2}>
+            <TextField
+              fullWidth
+              label="Number of returned queries"
+              type="number"
+              size="medium"
+              variant="outlined"
+              value={topN}
+              error={!!(topN && topN < 1)}
+              helperText={topN && topN < 1 ? "Number must be bigger than zero" : " "}
+              onChange={onTopNChange}
+              onKeyDown={onKeyDown}
+            />
+          </Box>
+          <Box>
+            <Tooltip title="Apply">
+              <IconButton onClick={onApplyQuery} sx={{height: "49px", width: "49px"}}>
+                <PlayCircleOutlineIcon/>
+              </IconButton>
+            </Tooltip>
+          </Box>
+        </Box>
+        <Typography variant="body1" pt={2}>
+            VictoriaMetrics tracks the last&nbsp;
+          <Tooltip arrow title={<Typography>search.queryStats.lastQueriesCount</Typography>}>
+            <b style={{cursor: "default"}}>
+              {data ? formatPrettyNumber(data["search.queryStats.lastQueriesCount"]) : "search.queryStats.lastQueriesCount"}
+            </b>
+          </Tooltip>
```

Comment:
i think we can define another function which we can call and build title. this will simplify part of code. And i think we can reuse it in the next title